src/main/java/seedu/address/model/person/Rsvp.java

```diff
@@ -0,0 +1,102 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents an RSVP status for a person.
+ * The RSVP status can be one of the following: "PENDING", "ACCEPTED", "DECLINED".
+ * This class enforces the following constraints:
+ * RSVP Status should be either "PENDING", "ACCEPTED", or "DECLINED".</li>
+ * The first character of the RSVP status must not be a whitespace.</li>
+ */
+public class Rsvp {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "RSVP Status should be 'ACCEPTED', 'DECLINED'";
+    private static final String[] RSVP_STATUS = {"PENDING", "ACCEPTED", "DECLINED"};
+
+    /**
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    private String rsvp;
+
+    /**
+     * Constructs an {@code Rsvp} object with the specified RSVP status.
+     *
+     * @param rsvp A valid RSVP status.
+     * @throws NullPointerException if the given rsvp is null.
+     * @throws IllegalArgumentException if the given rsvp is invalid.
+     */
+    public Rsvp(String rsvp) {
+        requireNonNull(rsvp);
+        checkArgument(isValidRsvp(rsvp), MESSAGE_CONSTRAINTS);
+        this.rsvp = rsvp;
+    }
```

Comment:
It looks good, but just a small thing would be maybe to standaridise how rsvp is being stored. Either all upper or lower cases so it looks the same when toString is called.

Reply:
Agreed, can call toUpperCase() perhaps.